### ISSUE TITLE
Yolov10 model gives different results on how to use..

### ISSUE THREAD

**Issue by gihyukcho1** [THU-MIG/yolov10#518]
:
import _init_paths
from ultralytics import YOLOv10
from ultralytics import YOLO
import torch
import torch.nn as nn
import cv2
import torchvision.transforms as transforms
import numpy as np

def image_to_tensor(image_path, target_size=(640, 640), orig_size=(384, 640)):

    img = cv2.imread(image_path)
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
    
    img = cv2.resize(img, (orig_size[1], orig_size[0]))
    
    # Create black canvas of target size
    canvas = np.zeros((target_size[0], target_size[1], 3), dtype=np.uint8)
    
    # Calculate padding
    y_offset = (target_size[0] - orig_size[0]) // 2
    x_offset = (target_size[1] - orig_size[1]) // 2
    
    # Place image in center
    canvas[y_offset:y_offset+orig_size[0], 
           x_offset:x_offset+orig_size[1]] = img
    # Preprocessing
    transform = transforms.Compose([
        transforms.ToTensor(),
        transforms.Normalize(mean=[0,0,0], std=[1,1,1]),
    ])
    
    # Convert to tensor and add batch dimension
    tensor = transform(canvas).unsqueeze(0)
    return tensor.cuda(), canvas

model = YOLOv10("../runs/detect/train9/weights/best.pt")
model.cuda()

if __name__ == "__main__":
    # Load image as tensor
    image_tensor, canvas = image_to_tensor("/home/gihyukcho/Dropbox/label_ws/raw_data/2025-02-20/left/000125.png")
   
    result = model(image_tensor)
    result2 = model.predict("/home/gihyukcho/Dropbox/label_ws/raw_data/2025-02-20/left/000125.png", save=True, imgsz=640, conf=0.15)
    print("Result 1")
    print(result[0].boxes.data)
    print("Result 3")
    print(result2[0].boxes.data)

The above is my code. I found that just using Yolov10(image_tensor) and Yolov10.predict have different results. How to make them coincident?
My original image has 1920*1080 shape. This is resized into 640*384 and the remain part is padded by black(0,0,0) to make 640*640 image.  And then inserted into Yolov10. But it is different from using .predict method. Why? and How to fix?

Thanks in advance..
